### ISSUE TITLE
Bring back adjustable Zoom/Aspect Ratio

### ISSUE THREAD

**Issue by psit** [FreeRDP/Remmina#937]
Hi there!

Old versions of Remmina allowed to set a variable zoom-factor and optionally fixed aspect ratios when zooming (scaling x-/y-axis independently or locked). This feature was extremely helpful when working via VNC on big screens or even dual/triple monitors (while e.g. using a small laptop screen).
Sometimes neither scale-to-fit, nor unscaled are viable options and "something in between" would be very handy. I miss this feature in the new versions of remmina. 

Kind regards,
Peter

**Comment by nixomose** [comment]
Same for me.

**Comment by wrffrz** [comment]
Yep, the current set of scaling options make this tool completely unusable when connecting to my dual-monitor desktop from my laptop.

**Comment by antenore** [comment]
I cannot remember about this feature, was it a profile level, general setting? Was it only for VNC? I remember about 2 horizontal sliders but I'm not sure about it.

Any clue starting from which version we have removed it?

Beside this, probably I've deleted all of that part while migrating from Gtk+ 2 and Gtk+ 3.

I remember something around that sliders that was not possible to migrate.

Another thing is that fractional scaling are not (yet) well supported by most desktops. There are some discussion around Gnome3 on going.

BTW, please help me to find everything I need to do a proper analysis and eventually come back with a fix. Thanks!

**Comment by psit** [comment]
Great that you want to work on it.
A quick image search on "remmina scale" shows these two screenshots that show where the feature was available:
1. Advanced Settings (which is not so important in my personal opinion)
![link to screenshot](https://canne.files.wordpress.com/2012/07/remmina_n9_vnc_session_config-2.png)
2. Using the toolbar changing the view-mode
![link to screenshot](http://grdc.sourceforge.net/screenshot5.png)

Looking at the sources of the images, you can see that the feature was implemented in the "original" grdc project back in 2009. 
I'm pretty sure the feature was working in ubuntu 10.04 lucid which would have been remmina 0.7.4
I'm not sure if it was still there in precise 12.04 with remmina 1.0

**Comment by antenore** [comment]
Ah! Yes! Now I remember. It was a nightmare to implement in GTK3.

We still have the remmina "master" branch that is used by most of debian based distros. 

[Here](https://github.com/FreeRDP/Remmina/blob/master/remmina-plugins/vnc/vnc_plugin.c#L1958) it is.

So I can at least investigate.

**Comment by jef-n** [comment]
BTW not only worked in VNC, but also in RDP.

**Comment by ralfthewise** [comment]
I just submitted a PR (#1377) that at least allows you to zoom to fit in the existing window, that may be sufficient for most people's use cases.  In the interim, just a heads up that you can get this behavior by manually editing the config file for a connection:

```
[tim@hood:~/dev/Remmina (next)]$ cat /home/tim/.local/share/remmina/1506450447354.remmina 
[remmina]
keymap=
ssh_auth=0
quality=1
disableencryption=0
postcommand=
ssh_privatekey=
disableclipboard=0
ssh_charset=
group=
name=whatever
ssh_username=
precommand=
viewonly=0
proxy=
ssh_loopback=0
colordepth=16
protocol=VNC
ssh_server=
ssh_enabled=0
username=
disablepasswordstoring=0
server=whatever
showcursor=0
disableserverinput=0
resolution_width=1920
resolution_height=1080
viewmode=4
window_maximize=0
keyboard_grab=1
window_height=917
window_width=1086
scale=1                                      <== add this line
dynamic_resolution_height=0
dynamic_resolution_width=0
```

**Comment by ic3d** [comment]
Enhancement to be considered in the near future.

**Comment by antenore** [comment]
WARNING:
The Remmina project is moving to gitlab.com, in order to keep receiving notification about this issue,
you should have a gitlab account and link it to github.
The migration will happen in the next days.
Thanks for your support and understanding.

cc @antenore @ic3d @jef-n @nixomose @psit @ralfthewise @wrffrz
